lib/racecar/instrumenter.rb

```diff
@@ -1,3 +1,5 @@
+require "concurrent/utility/monotonic_time"
```

Comment:
What does this do? I don't think we want a dependency on concurrent-ruby from Racecar.

Reply:
When using Racecar with Ruby versions > 2.6 (at least based on my tests, might impact some previous versions as well) I was getting `NoMethodError: undefined method `monotonic_time' for Concurrent:Module`. The above line solves the problem. It's not really related to the changes introduced by this PR but I had to add it because we're running our apps with newer versions of Ruby. I'm not sure with which Ruby versions the Racecar team usually tests the library.

Reply:
I believe that dependency was removed in `master`. Can you rebase and see if it's still a problem?

Reply:
Just checked, my branch was already up to date:

```
➜  racecar git:(master) git log -n 5 --oneline
19f997f (HEAD -> master) Track number of retries when handling errors
0a821e1 Avoid undefined method `monotonic_time' for Concurrent:Module
b614736 (upstream/master, origin/master, origin/HEAD) Merge pull request #185 from zendesk/dasch-allow-nil-keys
0114e4a Allow `key:` to be nil in `#produce`
97d6377 Merge pull request #183 from zendesk/dasch-docker-compose
```

I tried removing the `require` and running the specs and got:

```
Racecar::Instrumenter
  #instrument
    applies a namespace
    appends a default payload

running a Racecar consumer
=> Starting Racecar consumer EchoConsumer...
=> Wrooooom!
=> Ctrl-C to shutdown consumer
=> Crashed: NoMethodError: undefined method `monotonic_time' for Concurrent:Module
/Users/cmarques/.rbenv/versions/2.7.1/lib/ruby/gems/2.7.0/gems/activesupport-6.0.3.3/lib/active_support/notifications/instrumenter.rb:140:in `now'
/Users/cmarques/.rbenv/versions/2.7.1/lib/ruby/gems/2.7.0/gems/activesupport-6.0.3.3/lib/active_support/notifications/instrumenter.rb:79:in `start!'
/Users/cmarques/.rbenv/versions/2.7.1/lib/ruby/gems/2.7.0/gems/activesupport-6.0.3.3/lib/active_support/subscriber.rb:132:in `start'
/Users/cmarques/.rbenv/versions/2.7.1/lib/ruby/gems/2.7.0/gems/activesupport-6.0.3.3/lib/active_support/notifications/fanout.rb:156:in `start'
/Users/cmarques/.rbenv/versions/2.7.1/lib/ruby/gems/2.7.0/gems/activesupport-6.0.3.3/lib/active_support/notifications/fanout.rb:58:in `block in start'
/Users/cmarques/.rbenv/versions/2.7.1/lib/ruby/gems/2.7.0/gems/activesupport-6.0.3.3/lib/active_support/notifications/fanout.rb:58:in `each'
/Users/cmarques/.rbenv/versions/2.7.1/lib/ruby/gems/2.7.0/gems/activesupport-6.0.3.3/lib/active_support/notifications/fanout.rb:58:in `start'
/Users/cmarques/.rbenv/versions/2.7.1/lib/ruby/gems/2.7.0/gems/activesupport-6.0.3.3/lib/active_support/notifications/instrumenter.rb:36:in `start'
/Users/cmarques/.rbenv/versions/2.7.1/lib/ruby/gems/2.7.0/gems/activesupport-6.0.3.3/lib/active_support/notifications/instrumenter.rb:22:in `instrument'
/Users/cmarques/.rbenv/versions/2.7.1/lib/ruby/gems/2.7.0/gems/activesupport-6.0.3.3/lib/active_support/notifications.rb:180:in `instrument'
/Users/cmarques/stuff/racecar/lib/racecar/instrumenter.rb:21:in `instrument'
/Users/cmarques/stuff/racecar/lib/racecar/runner.rb:65:in `block in run'
/Users/cmarques/stuff/racecar/lib/racecar/runner.rb:62:in `loop'
/Users/cmarques/stuff/racecar/lib/racecar/runner.rb:62:in `run'
/Users/cmarques/stuff/racecar/lib/racecar.rb:51:in `run'
/Users/cmarques/stuff/racecar/lib/racecar/cli.rb:63:in `run'
/Users/cmarques/stuff/racecar/spec/integration/consumer_spec.rb:42:in `block (3 levels) in <top (required)>'
#<Thread:0x00007f81289c1070 /Users/cmarques/stuff/racecar/spec/integration/consumer_spec.rb:40 run> terminated with exception (report_on_exception is true):
Traceback (most recent call last):
        16: from /Users/cmarques/stuff/racecar/spec/integration/consumer_spec.rb:42:in `block (3 levels) in <top (required)>'
        15: from /Users/cmarques/stuff/racecar/lib/racecar/cli.rb:63:in `run'
        14: from /Users/cmarques/stuff/racecar/lib/racecar.rb:51:in `run'
```

Reply:
Hmm, that looks like a transitive dependency through ActiveSupport. Can you move the `require` to where we require the ActiveSupport stuff, so it only happens there, and add a comment describing the reason?

Reply:
@dasch This is interesting. The problem seems to be related to some interaction/things being pre-loaded when running the whole test suite in one go. If I run the unit tests and the integration tests separately, like the CircleCI build does, then everything works. I removed the commit adding `require "concurrent/utility/monotonic_time"` since removing the interaction problem introduced by running `bundle exec rspec spec` is outside the scope of this PR.

Thanks for reviewing and challenging this!

Reply:
:+1:

Perhaps remove the change from this PR for now then? I'm interested in getting to the bottom of this, but it's a separate issue from this PR.

Reply:
Ah, you already did that. Thanks!